storage/src/nodestore/mod.rs

```diff
@@ -350,11 +354,27 @@ pub trait RootReader {
 }
 
 /// A committed revision of a merkle trie.
-#[derive(Clone, Debug)]
+#[derive(Debug)]
 pub struct Committed {
     deleted: Box<[MaybePersistedNode]>,
     root_hash: Option<TrieHash>,
     root: Option<MaybePersistedNode>,
+    /// TODO: No readers of this variable yet - will be used for tracking unwritten nodes in committed revisions
+    unwritten_nodes: AtomicUsize,
+}
+
+impl Clone for Committed {
+    fn clone(&self) -> Self {
+        Self {
+            deleted: self.deleted.clone(),
+            root_hash: self.root_hash.clone(),
+            root: self.root.clone(),
+            unwritten_nodes: AtomicUsize::new(
+                self.unwritten_nodes
+                    .load(std::sync::atomic::Ordering::Relaxed),
+            ),
```

Comment:
Confirming that this is intentional the the cloned committed is using a new atomic integer and isn't re-using the integer from the previous cloned. The diverging committed nodes will have different unwritten_nodes counts.